app/src/test/java/org/simple/clinic/registration/phone/RegistrationPhoneScreenControllerTest.kt

```diff
@@ -48,32 +49,23 @@ class RegistrationPhoneScreenControllerTest {
   private val userUuid = UUID.fromString("a5c55e97-dcad-4cd8-9832-4da9f7b3d4b7")
   private val defaultOngoingEntry = OngoingRegistrationEntry(uuid = userUuid)
 
-  private val controller: RegistrationPhoneScreenController = RegistrationPhoneScreenController(
-      userSession = userSession,
-      userLookup = findUserWithPhoneNumber,
-      numberValidator = numberValidator,
-      facilitySync = facilitySync,
-      uuidGenerator = FakeUuidGenerator.fixed(userUuid)
-  )
-
-  @Before
-  fun setUp() {
-    whenever(userSession.isOngoingRegistrationEntryPresent())
-        .doReturn(Single.never())
-    whenever(userSession.isUserUnauthorized())
-        .doReturn(Observable.never())
-
-    uiEvents
-        .compose(controller)
-        .subscribe { uiChange -> uiChange(screen) }
+  private lateinit var controller: RegistrationPhoneScreenController
+
+  private lateinit var controllerSubscription: Disposable
+
+  @After
+  fun tearDown() {
+    controllerSubscription.dispose()
   }
 
   @Test
   fun `when screen is created and an existing ongoing entry is absent then an empty ongoing entry should be created`() {
     whenever(userSession.saveOngoingRegistrationEntry(defaultOngoingEntry)).doReturn(Completable.complete())
     whenever(userSession.isOngoingRegistrationEntryPresent()).doReturn(Single.just(false))
     whenever(userSession.ongoingRegistrationEntry()).doReturn(Single.never())
+    whenever(userSession.isUserUnauthorized()).doReturn(Observable.never())
```

Comment:
Why did we start adding new `whenever ` conditions in this commit? is it because we have started setting up controller for every test? 🤔

Reply:
Yes, in `setupController` we triggered the `ScreenCreated` event so we had to set up stubs in all the tests.